plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -110,7 +110,11 @@ func (a *podSecurityPolicyPlugin) SetInternalKubeInformerFactory(f informers.Sha
 // 3.  Try to generate and validate a PSP with providers.  If we find one then admit the pod
 //     with the validated PSP.  If we don't find any reject the pod and give all errors from the
 //     failed attempts.
-func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
+func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) (err error) {
+	defer func() {
+		ObserveAdmit(err != nil, a)
```

Comment:
This doesn't look correct.  Just considering the final return of this function, won't this give your the wrong value for rejected?

Reply:
I guess maybe I don't understand what the final value is.  I'm expecting err to be a particular instance that may or may not be the one returned, so the final return value isn't reflected in the value of err at the point at which you're called.

I'm not new to golang and I'd say that if I don't know the answer right off, this code is rather confusing.

Reply:
The named return value is always going to be the final return (as long as it's not shadowed), but since `err` is often shadowed I can see how this would be confusing. I'll wrap an internal admit to make it more clear.